common/src/main/java/io/netty5/util/concurrent/SystemTicker.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.util.concurrent;
+
+import java.util.concurrent.TimeUnit;
+
+enum SystemTicker implements Ticker {
+    INSTANCE;
+
+    static final long START_TIME = System.nanoTime();
+
+    @Override
+    public long initialNanoTime() {
+        return START_TIME;
+    }
+
+    @Override
+    public long nanoTime() {
+        return System.nanoTime() - START_TIME;
+    }
+
+    @Override
+    public void sleep(long delay, TimeUnit unit) throws InterruptedException {
```

Comment:
Add `Objects#requireNonNull` for `TimeUnit` because performing an operation on `null` will result in NPE.